game/src/contrib/systems/HealthbarSystem.java

```diff
@@ -0,0 +1,92 @@
+
+import contrib.components.HealthComponent;
+
+import core.Entity;
+import core.System;
+import core.components.PositionComponent;
+import core.components.UIComponent;
+import core.systems.CameraSystem;
+import core.utils.Point;
+import core.utils.logging.CustomLogLevel;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.logging.Logger;
+
+public final class HealthbarSystem extends System {
+    private static final Logger LOGGER = Logger.getLogger(HealthbarSystem.class.getSimpleName());
+
+    private final Map<Integer, ProgressBar> healthbarMapping = new HashMap<>();
+
+    public HealthbarSystem() {
+        super(HealthComponent.class, PositionComponent.class);
+        this.onEntityAdd =
+                (x) -> {
+                    LOGGER.log(CustomLogLevel.TRACE, "HealthbarSystem got send a new Entity");
+                    ProgressBar newHealthbar =
+                            createNewHealthbar(x.fetch(PositionComponent.class).orElseThrow());
+                    LOGGER.log(CustomLogLevel.TRACE, "created a new Healthbar");
+                    Entity e = new Entity();
+                    LOGGER.log(CustomLogLevel.TRACE, "created a new Entity for the Healthbar");
+                    new UIComponent(e, new Container<>(newHealthbar), false);
+                    LOGGER.log(CustomLogLevel.TRACE, "created a new UIComponent for the Healthbar");
+                    healthbarMapping.put(x.id(), newHealthbar);
+                    LOGGER.log(CustomLogLevel.TRACE, "HealthbarSystem added to temporary mapping");
+                };
+        LOGGER.log(CustomLogLevel.TRACE, "HealthbarSystem onEntityAdd was changed");
+        this.onEntityRemove = (x) -> healthbarMapping.remove(x.id()).remove();
+        LOGGER.log(CustomLogLevel.TRACE, "HealthbarSystem onEntityRemove was changed");
+        LOGGER.info("HealthbarSystem created");
+    }
+
+    @Override
+    public void execute() {
+        entityStream().map(this::buildDataObject).forEach(this::update);
+    }
+
+    private void update(EnemyData ed) {
+        // entferne wenn tod
+        if (ed.hc.currentHealthpoints() <= 0) ed.pb.remove();
+        // set visible only if entity lost health
+        // ed.pb.setVisible(ed.hc.currentHealthpoints() != ed.hc.maximalHealthpoints());
+        updatePosition(ed.pb, ed.pc);
+
+        // set value to health percent
+        ed.pb.setValue((float) ed.hc.currentHealthpoints() / ed.hc.maximalHealthpoints() * 100);
+    }
+
+    private EnemyData buildDataObject(Entity entity) {
+        return new EnemyData(
+                entity.fetch(HealthComponent.class).orElseThrow(),
+                entity.fetch(PositionComponent.class).orElseThrow(),
+                healthbarMapping.get(entity.id()));
+    }
+
+    private ProgressBar createNewHealthbar(PositionComponent pc) {
+        ProgressBar.ProgressBarStyle healthstyle =
+                HeroUISystem.createNewPBStyleWhichShouldBeInAtlasAndIsAToDoYesItIsUglyToAnnoyAll(
```

Comment:
griffiger name

Reply:
Diese gehört eigentlich aufgelöst, sobald die Diskussion in #865 behoben wurde oder gemerged wurde. Hier werden Texturen zur Laufzeit abgeändert und einfach nur eine Färbung angewandt. Dies würde entweder in den Skin wandern(in den Atlas) oder in eine nicht kompilierte Konfiguration. 

Da der Atlas wie in #865 bereits erwähnt jedes Mal, wenn eine weitere Texture hinzugefügt wird, komplett neu erstellt wird und somit eigentlich für eine Versionierung nicht geeignet ist. Sind hier aktuell die Änderungen im Code und sobald die Entscheidung getroffen wurde, würde ich die Methode löschen und in den Skin übertragen.

Reply:
Weil ich grade drüber Stolper: Das machst du dann nach dem Atlas-PR?